pallets/subtensor/src/swap.rs

```diff
@@ -0,0 +1,422 @@
+        let stake_count = stakes.len() as u32;
+        for (coldkey, stake_amount) in stakes {
+            Stake::<T>::insert(new_hotkey, &coldkey, stake_amount);
+            let _ = Stake::<T>::clear_prefix(old_hotkey, stake_count, None);
+            writes += 2; // One write for insert and one for remove
+        }
+
+        *weight += T::DbWeight::get().writes(writes as u64);
+        Ok(())
+    }
+    /// Swaps the network membership status of the hotkey.
+    ///
+    /// # Arguments
+    ///
+    /// * `old_hotkey` - The old hotkey.
+    /// * `new_hotkey` - The new hotkey.
+    /// * `netuid_is_member` - A vector of network IDs where the hotkey is a member.
+    /// * `weight` - The weight of the transaction.
+    ///
+    /// # Returns
+    ///
+    /// * `Result<(), Error<T>>` - The result of the operation.
+    pub fn swap_is_network_member(
+        old_hotkey: &T::AccountId,
+        new_hotkey: &T::AccountId,
+        netuid_is_member: &[u16],
+        weight: &mut Weight,
+    ) -> DispatchResult {
+        let _ = IsNetworkMember::<T>::clear_prefix(old_hotkey, netuid_is_member.len() as u32, None);
+        weight.saturating_accrue(T::DbWeight::get().writes(netuid_is_member.len() as u64));
+        for netuid in netuid_is_member.iter() {
+            IsNetworkMember::<T>::insert(new_hotkey, netuid, true);
+            weight.saturating_accrue(T::DbWeight::get().writes(1));
+        }
+        Ok(())
+    }
+
+    /// Swaps the axons of the hotkey.
+    ///
+    /// # Arguments
+    ///
+    /// * `old_hotkey` - The old hotkey.
+    /// * `new_hotkey` - The new hotkey.
+    /// * `netuid_is_member` - A vector of network IDs where the hotkey is a member.
+    /// * `weight` - The weight of the transaction.
+    ///
+    /// # Returns
+    ///
+    /// * `Result<(), Error<T>>` - The result of the operation.
+    pub fn swap_axons(
+        old_hotkey: &T::AccountId,
+        new_hotkey: &T::AccountId,
+        netuid_is_member: &[u16],
+        weight: &mut Weight,
+    ) -> DispatchResult {
+        for netuid in netuid_is_member.iter() {
+            if let Ok(axon_info) = Axons::<T>::try_get(netuid, old_hotkey) {
+                Axons::<T>::remove(netuid, old_hotkey);
+                Axons::<T>::insert(netuid, new_hotkey, axon_info);
+                weight.saturating_accrue(T::DbWeight::get().writes(2));
+            }
+        }
+        Ok(())
+    }
+    /// Swaps the references in the keys storage map of the hotkey.
+    ///
+    /// # Arguments
+    ///
+    /// * `old_hotkey` - The old hotkey.
+    /// * `new_hotkey` - The new hotkey.
+    /// * `netuid_is_member` - A vector of network IDs where the hotkey is a member.
+    /// * `weight` - The weight of the transaction.
+    ///
+    /// # Returns
+    ///
+    /// * `Result<(), Error<T>>` - The result of the operation.
+    pub fn swap_keys(
+        old_hotkey: &T::AccountId,
+        new_hotkey: &T::AccountId,
+        netuid_is_member: &Vec<u16>,
```

Comment:
Maybe better to just accept a `&[u16]` instead of a `Vec`.

Reply:
done